### ISSUE TITLE
Got this error using GAE

### ISSUE THREAD

**Issue by paulogr** [cloudinary/pycloudinary#52]
Anyone can help me ?

Thanks

Encountered unexpected error from ProtoRPC method implementation: TypeError (<type 'instance'> has type <type 'instance'>, but expected one of: str, unicode)
Traceback (most recent call last):
  File "/base/data/home/runtimes/python27/python27_lib/versions/third_party/protorpc-1.0/protorpc/wsgi/service.py", line 181, in protorpc_service_app
    response = method(instance, request)
  File "/base/data/home/runtimes/python27/python27_lib/versions/third_party/endpoints-1.0/endpoints/api_config.py", line 1332, in invoke_remote
    return remote_method(service_instance, request)
  File "/base/data/home/runtimes/python27/python27_lib/versions/third_party/protorpc-1.0/protorpc/remote.py", line 412, in invoke_remote_method
    response = method(service_instance, request)
  File "/base/data/home/apps/s~sprtsapp-api/1.385327187522572524/sprtsapp_api.py", line 202, in upload_cover
    upload_result = upload(request.file)
  File "/base/data/home/apps/s~sprtsapp-api/1.385327187522572524/lib/cloudinary/uploader.py", line 16, in upload
    return call_api("upload", params, file = file, *_options)
  File "/base/data/home/apps/s~sprtsapp-api/1.385327187522572524/lib/cloudinary/uploader.py", line 199, in call_api
    response = urllib2.urlopen(request, *_kw).read()
  File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 404, in open
    response = self._open(req, data)
  File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 422, in _open
    '_open', req)
  File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(_args)
  File "/base/data/home/apps/s~sprtsapp-api/1.385327187522572524/lib/cloudinary/poster/streaminghttp.py", line 172, in https_open
    return self.do_open(StreamingHTTPSConnection, req)
  File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 1189, in do_open
    r = h.getresponse()
  File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/gae_override/httplib.py", line 522, in getresponse
    *_extra_kwargs)
  File "/base/data/home/runtimes/python27/python27_lib/versions/1/google/appengine/api/urlfetch.py", line 270, in fetch
    allow_truncated, follow_redirects, validate_certificate)
  File "/base/data/home/runtimes/python27/python27_lib/versions/1/google/appengine/api/urlfetch.py", line 326, in make_fetch_call
    request.set_payload(payload)
  File "cpp_message.pyx", line 124, in cpp_message.SetScalarAccessors.Setter (third_party/apphosting/python/protobuf/proto1/cpp_message.cc:2229)
TypeError: <type 'instance'> has type <type 'instance'>, but expected one of: str, unicode

**Comment by taragano** [comment]
Please have a look at our [GAE example project](https://github.com/cloudinary/pycloudinary/tree/master/samples/gae). 
Specifically: 
https://github.com/cloudinary/pycloudinary/blob/master/samples/gae/main.py#L38

**Comment by paulogr** [comment]
After a lot of tries I realized that the function to check if its on GAE doesn't work

https://github.com/cloudinary/pycloudinary/blob/master/cloudinary/uploader.py#L230

Always return false to me.

Can you help us with this ?

**Comment by taragano** [comment]
Please note that GAE only works with python 2.7.
Perhaps you use python 3 or have it also installed on the same machine?
Also, please make sure you use to the latest version of our library.

**Comment by paulogr** [comment]
Hi,
I did clone the master version of repo, and sure I'm using Python 27.

Its specified on my app.yaml

Em qua, 1 de jul de 2015 às 04:03, Itay Taragano <notifications@github.com>
escreveu:

**Comment by taragano** [comment]
Can you please add some logging in each branch of the if/else?
This would be helpful to debug.

**Comment by taragano** [comment]
Closing this issue due to the time elapsed.
Please feel free to contact our support http://support.cloudinary.com or create a new ticket if you have any additional issues.

**Comment by danmcclure** [comment]
I have the same problem.  cloudinary.uploader.add_tag() crashes on gae with the same message. But works on local dev server.  I appreciate your support for gae!  Thank you.  Would love to get this to work.  For now I'm using cloudinary.api.update().

"""
            #
            # @TODO: crashes on gae, ok on dev server
            #
            pub_ids = [
                        stat.image_orig_url,
                        stat.image_meme_url
                       ]

            if stat.image_rwd_url:  pub_ids.extend([stat.image_rwd_url])

            cloudinary.uploader.add_tag("s_%s" % stat_id, pub_ids)
            cloudinary.uploader.remove_tag("pending", pub_ids )
            """

            tags_orig.remove('pending')
            tags_meme.remove('pending')
            tags_rwd.remove('pending')
            tags_orig.append("s_%s" % stat_id)
            tags_meme.append("s_%s" % stat_id)
            tags_rwd.append("s_%s" % stat_id)

            cloudinary.api.update(stat.image_orig_url, tags=tags_orig)
            cloudinary.api.update(stat.image_meme_url, tags=tags_meme)

            if stat.image_rwd_url: 
                cloudinary.api.update(stat.image_rwd_url, tags=tags_rwd)

**Comment by deems** [comment]
I have the same problem. Here is my code:
```
file = self.request.get('photo')
str_file = StringIO.StringIO(file)
str_file.name = 'file'
upload_result = cloudinary.uploader.upload(str_file)
```
Always I get this error:
``File "/base/data/home/apps/s~banded-earth-605/122.393683277654675263/imager.py", line 223, in post upload_result = cloudinary.uploader.upload(str_file) File "/base/data/home/apps/s~banded-earth-605/122.393683277654675263/lib/cloudinary/uploader.py", line 16, in upload return call_api("upload", params, file = file, **options) File "/base/data/home/apps/s~banded-earth-605/122.393683277654675263/lib/cloudinary/uploader.py", line 200, in call_api response = urllib2.urlopen(request, **kw).read() File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 127, in urlopen return _opener.open(url, data, timeout) File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 404, in open response = self._open(req, data) File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 422, in _open '_open', req) File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 382, in _call_chain result = func(*args) File "/base/data/home/apps/s~banded-earth-605/122.393683277654675263/lib/cloudinary/poster/streaminghttp.py", line 172, in https_open return self.do_open(StreamingHTTPSConnection, req) File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/urllib2.py", line 1189, in do_open r = h.getresponse() File "/base/data/home/runtimes/python27/python27_dist/lib/python2.7/gae_override/httplib.py", line 522, in getresponse **extra_kwargs) File "/base/data/home/runtimes/python27/python27_lib/versions/1/google/appengine/api/urlfetch.py", line 270, in fetch allow_truncated, follow_redirects, validate_certificate) File "/base/data/home/runtimes/python27/python27_lib/versions/1/google/appengine/api/urlfetch.py", line 326, in make_fetch_call request.set_payload(payload) File "cpp_message.pyx", line 124, in cpp_message.SetScalarAccessors.Setter (third_party/apphosting/python/protobuf/proto1/cpp_message.cc:2229) TypeError: <type 'instance'> has type <type 'instance'>, but expected one of: str, unicode`

**Comment by deems** [comment]
Well, I fixed it #70
